packages/php/email-editor/src/Engine/class-email-editor.php

```diff
@@ -282,16 +282,18 @@ public function load_email_preview_template( string $template ): string {
 
 		$email_post_types = array_column( $this->get_post_types(), 'name' );
 
-		if ( in_array( $current_post_type, $email_post_types, true ) && locate_template( array( "single-$current_post_type.php" ) ) !== $template ) {
-			/*
-			 * This is an email post
-			 * AND a 'single $post_type template' is not found on
-			 * theme or child theme directories, so load it
-			 * from our plugin directory.
-			 */
-			return __DIR__ . '/Templates/single-post-template.php';
+		if ( ! in_array( $current_post_type, $email_post_types, true ) ) {
+			return $template;
 		}
 
-		return $template;
+		add_filter(
```

Comment:
This new filter seems a bit hidden in this place. I would prefer having it in the `initialize` method like the others. What do you think?

Reply:
This is because we need to use the [$post](https://github.com/mailpoet/mailpoet/blob/c15c4e67514f000d6c0ba78489aef9a11f87cc6e/packages/php/email-editor/src/Engine/class-email-editor.php#L279) variable available in this [scope](https://github.com/mailpoet/mailpoet/blob/c15c4e67514f000d6c0ba78489aef9a11f87cc6e/packages/php/email-editor/src/Engine/class-email-editor.php#L279-L295). 

I discovered a weird bug when testing with some plugins. The post from the [apply_filter](https://github.com/mailpoet/mailpoet/blob/c15c4e67514f000d6c0ba78489aef9a11f87cc6e/packages/php/email-editor/src/Engine/Templates/single-email-post-template.php#L10) gets overwritten. 

Also, we sometimes experience weird inconsistencies with some platforms (e.g., Atomic), so it's safer to use the variable here.